django/contrib/postgres/search.py

```diff
@@ -293,10 +293,28 @@ def __init__(self, expression, string, **extra):
         super().__init__(expression, string, **extra)
 
 
+class TrigramWordBase(Func):
+    output_field = FloatField()
+
+    def __init__(self, string, expression, **extra):
+        if not hasattr(string, 'resolve_expression'):
+            string = Value(string)
+        super().__init__(string, expression, **extra)
+
+
 class TrigramSimilarity(TrigramBase):
     function = 'SIMILARITY'
 
 
 class TrigramDistance(TrigramBase):
     function = ''
     arg_joiner = ' <-> '
+
+
+class TrigramWordDistance(TrigramWordBase):
+    function = ''
+    arg_joiner = ' <<-> '
+
+
+class TrigramWordSimilarity(TrigramWordBase):
+    function = 'WORD_SIMILARITY'
```

Comment:
I would use the same order as the previous two classes (`TrigramSimilarity` and `TrigramDistance`):

```suggestion
class TrigramWordSimilarity(TrigramWordBase):
    function = 'WORD_SIMILARITY'


class TrigramWordDistance(TrigramWordBase):
    function = ''
    arg_joiner = ' <<-> '
```

Reply:
@pauloxnet Ahh, sorry, just merged.